From bf210329cccc82a941e50e7341f919159c28fa87 Mon Sep 17 00:00:00 2001
From: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
Date: Tue, 14 Jun 2022 12:26:39 +0200
Subject: [PATCH] env: Fix prio on bad CRC case with ENV_APPEND enabled

Under no circumstances the environments with bad crc from the emmc are trusted.
The highest priority should always set to "0". See env_get_location() in
imx8m/soc.c for further information.

Signed-off-by: Michael Glembotzki <Michael.Glembotzki@iris-sensing.com>
---
 env/env.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/env/env.c b/env/env.c
index 7bb94dc8c1..9f30756157 100644
--- a/env/env.c
+++ b/env/env.c
@@ -191,11 +191,11 @@ int env_load(void)
 
 #if !CONFIG_IS_ENABLED(ENV_APPEND)
 			return 0;
-#endif
 		} else if (ret == -ENOMSG) {
 			/* Handle "bad CRC" case */
 			if (best_prio == -1)
 				best_prio = prio;
+#endif
 		} else {
 			debug("Failed (%d)\n", ret);
 		}
-- 
2.39.0

